app/services/katello/pulp3/repository_mirror.rb

```diff
@@ -91,10 +91,11 @@ def distribution_options(path, options = {})
 
       def remote_options
         base_options = common_remote_options
-        base_options.merge(url: remote_feed_url)
+        base_options.merge!(url: remote_feed_url)
         if (type_specific_options = repo_service.try(:mirror_remote_options))
```

Comment:
I think I was hitting this..Let me test with this change..I was seeing the remote issue on master branch..

Reply:
I'm not 100% convinced that it's a bug as long as the repository service sub-class provides a hash value. I would like folks to really tear down this change and if they agree it's a problem, we should change it. Otherwise, I'd rather keep the original implementation to avoid unneeded changes.

Reply:
Are you able to sync file repos to the proxy?
I am hitting this: 
```
Error: ArgumentError
The input argument (attributes) must be a hash in `PulpFileClient::FileFileRemote` initialize method
```
http://pastebin.test.redhat.com/911972

Reply:
I only tried yum. Does this happen on master?

Reply:
Oh, hey I wonder if the file repository service sub-class implements the mirror_repository_options?

Reply:
I started with files. Although my proxy is 3.6 pulpcore. The file plugin is up to date. If you are able to reproduce, that would be something we might want to fix for 3.6 as well?
Spinning up a 3.7 proxy to test the ansible piece.